dap_job_quality/utils/analysis_utils.py

```diff
@@ -0,0 +1,324 @@
+import numpy as np
+import pandas as pd
+import re
+
+
+def create_wide_table(df):
+    # Create a table with one row per ID, and a boolean column for each subcategory
+    dimensions_wide = df[["id", "subcategory"]].drop_duplicates()
+    dimensions_wide["value"] = 1
+    dimensions_wide = dimensions_wide.pivot(index="id", columns="subcategory").fillna(0)
+    dimensions_wide.columns = dimensions_wide.columns.get_level_values(1)
+    return dimensions_wide.reset_index()
+
+
+### HOURLY PAY ###
+
+
+def calculate_hourly_wage(
+    row, salary_unit_col="raw_salary_unit", hours_per_day=7.5, hours_per_year=37.5 * 52
+):
+    """Infer hourly pay.
+
+    If the pay is given per hour, accept this amount. Otherwise if it is given per day, divide by hours per day,
+    and if it is given per year, divide by the likely number of hours in a year.
+
+    Args:
+        row (_type_): _description_
+        hours_per_day (float, optional): _description_. Defaults to 7.5.
+        hours_per_year (_type_, optional): _description_. Defaults to 37.5*52.
+
+    Returns:
+        _type_: _description_
+    """
+    if row[salary_unit_col] == "hour":
+        return row["raw_salary_float"]
+    elif row[salary_unit_col] == "day":
+        return row["raw_salary_float"] / hours_per_day
+    elif row[salary_unit_col] == "year":
+        return row["raw_salary_float"] / hours_per_year
+    else:
+        return np.nan
+
+
+def extract_salary_info(text: str):
+    # Regular expression to match the salary pattern
+    salary_pattern = re.compile(
+        r"£\s*(\d+(?:,\d{3})*(?:\.\d{2})?)\s*(?:-|to)?\s*£?\s*(\d+(?:,\d{3})*(?:\.\d{2})?)?\s*(per\s*(hour|day|annum|year))?",
+        re.IGNORECASE,
+    )
```

Comment:
I'm sure this regex extracts most of the salaries successfully, but I guess it relies on some assumptions about how salaries are written and hasn't been evaluated (as far as I'm aware). So it doesn't require much (and you may have already done this), but I think it's worth making sure it's noted somewhere (e.g in a README somewhere) that the EY career salaries are extracted via certain assumptions. Otherwise it seems like the salaries are ground truth

Reply:
This is a really good point - thanks Liz